README.md
# Machinetron

## Getting Started

### Prerequisites

To run Team 2 Machinetron, you need to connect to the main control unit Raspberry Pi Zero through ssh. 

## Deployment

Follow the user manual to deploy Machinetron. The code to be run can be found through the Raspberry Pi Zero and the submachine boards shouldn't need to be touched until unless they need to be reprogrammed. To run the main code, save the desired STL file into the Machinetron folder. Inside the main.py file, inside line stlProcessor.generateCommands(), add your STL filename into the function input. Then run main.py inside terminal and press the GO button to start.

## Built with

[PyCharm] (https://www.jetbrains.com/pycharm/) - The IDE used to produce and run Python code

[Keil UVision] (http://www2.keil.com/mdk5/uvision/) - The IDE used to produce and run STM code

## Authors

* **Dion Lao** - *Python Raspberry Pi Code* - [DioNode] (https://github.com/dioNode)
* **Liam Chester** - *STM Microcontroller Code* - [LiamChester] (https://github.com/LiamChester)
* **Peter Hohl** - *STL Processing and Automation* - [peterhohl] (https://github.com/peterhohl)

history.txt
part0.STL
drill("front", 20.0, 25.2, 60)
drill("front", -20.0, 25.2, 60)
lathe(49, 79, 34.800000000000004)
controller.commandGenerator.retractMill(spinning=False)
millPointSequence([(38.2, 2.4), (38.1, 2.3), (38.1, 2.2), (37.6, 1.8), (37.6, 1.7), (36.6, 0.7), (36.6, 0.7), (36.2, 0.3), (36.1, 0.3), (35.8, 0.1), (38.2, 0.1)], 12, "top")
controller.commandGenerator.retractMill(spinning=True)
millPointSequence([(38.2, 2.4), (38.1, 2.3), (38.1, 2.2), (37.6, 1.8), (37.6, 1.7), (36.6, 0.7), (36.6, 0.7), (36.2, 0.3), (36.1, 0.3), (35.8, 0.1), (38.2, 0.1)], 20, "top")
controller.commandGenerator.retractMill(spinning=True)
controller.commandGenerator.retractMill(spinning=False)
millPointSequence([(-38.2, 2.4), (-38.2, 0.1), (-36.0, 0.1), (-36.2, 0.3), (-36.2, 0.3), (-36.6, 0.7), (-36.8, 0.7), (-37.8, 1.7), (-37.8, 1.8), (-38.2, 2.2), (-38.2, 2.3)], 12, "top")
controller.commandGenerator.retractMill(spinning=True)
millPointSequence([(-38.2, 2.4), (-38.2, 0.1), (-36.0, 0.1), (-36.2, 0.3), (-36.2, 0.3), (-36.6, 0.7), (-36.8, 0.7), (-37.8, 1.7), (-37.8, 1.8), (-38.2, 2.2), (-38.2, 2.3)], 20, "top")
controller.commandGenerator.retractMill(spinning=True)
controller.commandGenerator.retractMill(spinning=False)

config.py
configurationMap = {
    'handler': {
        'railSpeed': 10,
        'flipSpeed': 12,
        'spinSpeed': 100,
        'rapidRailSpeed': 50,
        'rapidSpinSpeed': 500,
        'initialRampUpSpeed': 100,
        'maxRailSpeed': 80,
        'maxFlipSpeed': 20,
        'maxSpinSpeed': 500,
        'railDPR': 39.23809524,
        'flipDPR': 25.7142857,
        'spinDPR': 360,
        'maxRail': 237.5,
        'maxFlip': 90,
        'id': 0x1A,
    }, 'drill': { # Location of lathe is measured relative to bottom
        'homeX': 205,
        'raiseSpeed': 15,
        'pushSpeed': 10,
        'spinSpeed': 500,
        'rapidRaiseSpeed': 25,
        'rapidPushSpeed': 50,
        'rapidSpinSpeed': 500,
        'maxRaiseSpeed': 30,
        'maxPushSpeed': 80,
        'maxSpinSpeed': 500,
        'raiseDPR': 8.06115358,
        'pushDPR': 31.643,
        'spinDPR': 1,
        'maxRaise': 140 - 5,
        'maxPush': 190,
        'diameter': 6,
        'depth': 80,
        'pushIncrement': 5,
        'id': 0x1F,
        'detectionTolerance': 10,
        'offsets': {
            'cuttingBit2HandlerCenter': 170,
            'cuttingBit2HandlerFlipBase': 134,
            'cuttingBitHeightOffset': 22,
            'cuttingBitHeightOffsetFlipped': 17,
            'motorStartDepthOffset': 15, # The amount of distance before the drill should start spinning
            'gripperSafetyVert': 10,
        },
    }, 'lathe': {
        'homeX': 91,
        'raiseSpeed': 2,
        'pushSpeed': 2,
        'spinSpeed': 0,
        'rapidRaiseSpeed': 25,
        'rapidPushSpeed': 50,
        'rapidSpinSpeed': 0,
        'maxRaiseSpeed': 30,
        'maxPushSpeed': 80,
        'maxSpinSpeed': 500,
        'raiseDPR': 8.06115358,
        'pushDPR': 31.643,
        'spinDPR': 1,
        'maxRaise': 145 - 10,
        'maxPush': 189,
        'length': 3,
        'depth': 60,
        'pushIncrement': 2,
        'id': 0x2A,
        'minDetectionRadius': 10,
        'maxDetectionRadius': 100,
        'offsets': {
            'cuttingBit2HandlerCenter': 189,
            'cuttingBit2HandlerFlipBase': 157,
            'cuttingBitHeightOffset': 19,
            'cuttingBitHeightOffsetFlipped': 11,
            'motorStartDepthOffset': 21, # The amount of distance before the handler should start spinning
            'gripperSafetyVert': 6,
        },
    }, 'mill': {
        'homeX': 46 + 7,
        'raiseSpeed': 10,
        'pushSpeed': 10,
        'spinSpeed': 500,
        'rapidRaiseSpeed': 25,
        'rapidPushSpeed': 50,
        'rapidSpinSpeed': 500,
        'maxRaiseSpeed': 40,
        'maxPushSpeed': 80,
        'maxSpinSpeed': 500,
        'raiseDPR': 8.06115358,
        'pushDPR': 31.643,
        'spinDPR': 1,
        'maxRaise': 129,
        'maxPush': 185,
        'diameter': 10,
        'depth': 60,
        'pushIncrement': 12,
        'id': 0x2F,
        'offsets': {
            'cuttingBit2HandlerCenter': 172,
            'cuttingBit2HandlerFlipBase': 136,
            'cuttingBitHeightOffset': 14,
            'cuttingBitHeightOffsetFlipped': 7,
            'motorStartDepthOffset': 30, # The amount of distance before the mill should start spinning
            'gripperSafetyVert': 10,
        },
    }, 'coordination': {
        'speed': 20,
    }, 'motorMap': {
        'spin': 2,
        'vert': 3,
        'pen': 1,
        'rail': 1,
        'flip': 3,
    }, 'instructions': {
        'NORM_INST': 0,
        'START_INST': 1,
        'PAUSE_INST': 2,
        'STOP_INST': 3,
    }, 'other': {
        'infVal': 9999,
        'homeVal': -10,
        'mmPerPixelRatio': 0.1, # Using the y direction
        'mmError': 1.0,
        'speedMultiplier': 10,
        'numRotationSteps': 200,
        'syncSpeed': 15,
        'gripperHeight': 20,
    }
}
